### DISCUSSION TITLE
Does Rust SDK automatically apply a "clock skew"?

### DISCUSSION THREAD

**Discussion by moranbw** [awslabs/aws-sdk-rust#1029] [Q&A] [answered]
It looks like some AWS SDKs have flags for a "clock skew", while some "automatically" apply this, like the v3 Javascript API.

I have a feeling that the Rust SDK does not?  I was having issues with a Windows user authenticating, getting persistent `SignatureDoesNotMatch` errors.  As a random guess, I asked if their clock was in sync.  They had manually set it.  I had them turn the time and timezone on to automatically set, and voila -- authentication worked immediately.

Is there any way I can programmatically account for this in my application?

**Comment by rcoh** [comment, marked as answer]
Interesting. The Rust SDK _does_ have middleware for correcting clock skew: https://github.com/smithy-lang/smithy-rs/blob/7eb008c4f7d852700e8c468fc1691e3cf880f2e7/aws/rust-runtime/aws-runtime/src/service_clock_skew.rs#L18

However, it looks like it's not used when computing the signature. I'm going to convert this to an issue to track the fix.
